apps/web/src/hooks/usePermit2Allowance.ts

```diff
@@ -0,0 +1,140 @@
+import { useInterval } from '@pancakeswap/hooks'
+import { CurrencyAmount, Token } from '@pancakeswap/sdk'
+import { PermitSignature, usePermitAllowance, useUpdatePermitAllowance } from 'hooks/usePermitAllowance'
+import { useCallback, useMemo, useState } from 'react'
+import { useHasPendingApproval, useHasPendingRevocation } from 'state/transactions/hooks'
+import { useAccount } from 'wagmi'
+import { ApprovalState, useApproveCallback } from './useApproveCallback'
+import useTokenAllowance from './useTokenAllowance'
+
+export enum AllowanceState {
+  LOADING,
+  REQUIRED,
+  ALLOWED,
+}
+
+export interface AllowanceRequired {
+  isRevocationPending: boolean
+  state: AllowanceState.REQUIRED
+  token: Token
+  isApprovalLoading: boolean
+  isApprovalPending: boolean
+  approve: any
+  permit: () => Promise<void>
+  revoke: any
+  needsSetupApproval: boolean
+  needsPermitSignature: boolean
+  allowedAmount: CurrencyAmount<Token>
+}
+
+export type Allowance =
+  | { state: AllowanceState.LOADING }
+  | {
+      state: AllowanceState.ALLOWED
+      permitSignature?: PermitSignature
+    }
+  | AllowanceRequired
+
+const AVERAGE_L1_BLOCK_TIME = 12000
+
+export default function usePermit2Allowance(
+  approvalAddress: string,
+  amount?: CurrencyAmount<Token>,
```

Comment:
yeah cool will do that too

Reply:
hmm after a second thought, let's keep it this way for now. It makes more sense to get the allowance for a token instead of a currency, as native currency doesn't need approval